src/librustc_middle/ty/mod.rs

```diff
@@ -196,21 +196,21 @@ pub struct AssocItem {
 
     /// Whether this is a method with an explicit self
     /// as its first argument, allowing method calls.
-    pub method_has_self_argument: bool,
+    pub fn_has_self_argument: bool,
```

Comment:
```suggestion
    pub fn_has_self_parameter: bool,
```
(And in the comment above.)

Reply:
Updated.